erpnext/stock/get_item_details.py

```diff
@@ -558,7 +558,7 @@ def get_price_list_rate_for(args, item_code):
 	if item_price_data:
 		if item_price_data[0][2] == args.get("uom"):
 			return item_price_data[0][1]
-		elif not args.get('price_list_uom_dependant'):
+		elif args.get('price_list_uom_dependant'):
```

Comment:
agreed the variable naming was incorrect, but this was defaulting to have price dependent on uom. Now this has to be set everywhere.

change price list doctype to have the variable named as `price_not_uom_dependent` 
(reference: https://www.grammar-monster.com/easily_confused/dependant_dependent.htm)

use the same variable everywhere instead of `price_list_uom_dependant`.

Change this block like so:
```
elif not args.get('price_not_uom_dependent'):
	return flt(item_price_data[0][1] * flt(args.get("conversion_factor", 1)))
else:
	return item_price_data[0][1]

```

Reply:
`Price List` DocType's field is already named `price_not_uom_dependant`
`get_item_details.py` sets a property (arg) named `price_list_uom_dependant`

So are you saying I should refactor `price_list_uom_dependant` in get_item_details to `not price_not_uom_dependant`

Reply:
No.
`price_list_uom_dependant` is actually the same as `price_not_uom_dependant` in Price List.

I'm asking you to change all instances of both these fields to the correct name:
`price_not_uom_dependent`

Subsequently, the existing logic was just fine.

Reply:
Follow this specific change in PR https://github.com/frappe/erpnext/pull/16544
I'll include the reason and Sagar's proposed changed in it.